Makefile

```
@@ -64,7 +64,14 @@ react-app-test: | $(REACT_APP_NODE_MODULES_PATH) react-app-lint
 	cd $(REACT_APP_PATH) && yarn test --no-watch --coverage
 
 .PHONY: test
+# If we only want to only test go code we have to change the test target
+# which is called by all.
+ifeq ($(GO_ONLY),1)
+test: common-test
+else
 test: common-test react-app-test
+endif
+
```

Rather than complicating things like this, split the test target.

```suggestion
test: common-test react-app-test

.PHONY: go-test
go-test:  common-test
```

Then in CI, you can call `make go-test` or `make react-app-test`.

---

Common-test only tests golang.

We need make-all: precheck style check_license lint yamllint unused build test

But make all calls test. We want it to test react locally, but only go in CI, hence the GO_ONLY.

---

So, we should change the `make` to not use the default `all` target. We expand what we want into separate steps.

---

Do we though? That just adds more bloat to the Makefiles:

```makefile
# Makefile
go-test: ...

test-go: precheck style check_license lint yamllint unused build go-test
```

```makefile
# Makefile.common
all: precheck style check_license lint yamllint unused build test
```

The idea of the `all` target is to generalize common tasks, maintaining an additional list of targets mostly the same is an annoyance. In my opinion, I think the current way is a lot easier to understand. It's just a condition that causes only go tests to run when `test` is called, as opposed to a whole new target which does mostly the same thing as the `all` target, except for the last step.

Running `all` on `test_go` follows a decision to classify the general tasks as part of Go, most of which are, but some (`yamllint`) don't exactly fit into that category. We may want to consider renaming the CircleCI job to just be `test` as it does run things that aren't strictly Go.

---

The final say is up to @roidelapluie.